### DISCUSSION TITLE
Regarding construction of functional data object

### DISCUSSION THREAD

**Discussion by shreyagg2202** [GAA-UAM/scikit-fda#457] [Q&A]
In the scikit-fda package, are there any functions available to create a functional data object using the coefficients and basis (B-spline basis) as compared to fd function in R.

**Comment by vnmabus** [comment]
Yes, the [`FDataBasis`](https://fda.readthedocs.io/en/stable/modules/autosummary/skfda.representation.basis.FDataBasis.html) object accepts a basis object (there are several: [`Monomial`](https://fda.readthedocs.io/en/stable/modules/autosummary/skfda.representation.basis.Monomial.html#skfda.representation.basis.Monomial), [`Fourier`](https://fda.readthedocs.io/en/stable/modules/autosummary/skfda.representation.basis.Fourier.html#skfda.representation.basis.Fourier), [`BSpline`](https://fda.readthedocs.io/en/stable/modules/autosummary/skfda.representation.basis.BSpline.html#skfda.representation.basis.BSpline), and you can even create your own) and a matrix of coefficients.

**Reply by shreyagg2202** [reply]
I want to represent the data as FData object which is an attribute to find the warping function, but not able to find any method which gives us FData object from my raw dataset.

**Reply by vnmabus** [reply]
`FData` is a superclass of both `FDataGrid` (discretized representation) and `FDataBasis` (basis expansion representation). You can create a `FDataGrid` from your dataset, and then use the `to_basis` method with a suitable basis, or use the `BasisSmoother` class to add a regularization term in the conversion process.

**Reply by shreyagg2202** [reply]
How can we find the warping function for a dataset whose time domain is not equal for all the functional data (curve).
I am also facing problem in creating FDataGrid with measurement of datasets in non equal time domains.

**Reply by vnmabus** [reply]
Unfortunately we currently don't support functional objects in different domains. You could probably try to pad the curves to a common length, align them using the methods in the registration module and then clip the padded parts.

**Reply by shreyagg2202** [reply]
We are trying to use clustering algorithm for a dataset which has several features as functional data, Like in the Canadian Weather Dataset having two functional features (Temperature and Precipitation). However I cannot find the clustering algorithms in scikit-fda using this kind of dataset, instead they only use a single functional data (curve).
How can we figure this out?

**Reply by vnmabus** [reply]
The $L_2$ distance works for vector-valued functions, so clustering algorithms that use this distance, such as k-means, should just work. I think that $L_{\infty}$ distance also works for `FDataGrid`. Implementing other $L_p$ distances should be easy to do.

In case that you mean two functional features with different domains, that is still not implemented. In the future we plan to implement these as pandas DataFrames with several functional data columns, but the distance functions still don't work with them.
